src/dxtbx/format/FormatNXmx.py

```diff
@@ -60,8 +60,8 @@ def _start(self):
     def _get_nxmx(self, fh: h5py.File):
         return dxtbx.nexus.nxmx.NXmx(fh)
 
-    def _beam(self, index=None):
-        return self._beam_model
+    def _beam(self, index: int = 0):
+        return self._beam_factory(index=index)
```

Comment:
Change function signature to reflect that `index` could be `None`
```suggestion
    def _beam(self, index: int | None = None):
        return self._beam_factory(index=index or 0)
```